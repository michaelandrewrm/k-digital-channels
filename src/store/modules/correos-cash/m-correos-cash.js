import BwipJs from 'bwip-js';
import SessionCache from '@modules/session/session-cache';

const cache = new SessionCache('correos-cash');

const normalizeModel = (model) => {
	const result = {
		beneficiaryProductNumber: model.destination.account.id,
		amount: model.amount.amount,
	};

	return result;
};

export default {
	namespaced: true,

	actions: {
		request({ dispatch }, model) {
			const payload = normalizeModel(model);
			const url = '/move-money/correos-cash/deposit';
			const method = 'POST';

			return dispatch(
				'service/request',
				{
					service: {
						request: { url, method },
					},
					payload,
				},
				{ root: true }
			).then(({ data }) => data);
		},

		get({ dispatch }, { paginationKey, depositId }) {
			if (depositId) {
				return cache.get(`item/${depositId}`);
			}

			const cacheKey = 'list/correos-cash';
			const queryParams = {};
			const url = '/move-money/correos-cash/deposits';
			const method = 'GET';

			if (cache.has(cacheKey) && !paginationKey) {
				return cache.get(cacheKey);
			}

			if (paginationKey) {
				Object.assign(queryParams, { paginationKey });
			}

			return dispatch(
				'service/request',
				{
					service: { request: { url, method } },
					queryParams,
				},
				{ root: true }
			).then(({ data }) => {
				data.data.forEach((item) => {
					if (item.details) {
						const details = item.details.map((qrData) => ({
							...qrData,
							beneficiary: item.beneficiary,
						}));

						Object.assign(item, { details });
					}

					cache.set(`item/${item.id}`, item);
				});

				if (cache.has(cacheKey)) {
					const result = cache.get(cacheKey);

					Object.assign(data.data, result.data.concat(data.data));
				}

				cache.set(cacheKey, data);

				return data;
			});
		},

		generateCodeData(context, data) {
			const xml = `
				<ORDEN>
					<OP>${data.operationId /* istanbul ignore next */ || ''}</OP>
					<CL>${data.clientId /* istanbul ignore next */ || ''}</CL>
					<IM>0</IM>
					<D1>${data.beneficiary.name /* istanbul ignore next */ || ''}</D1>
					<D2>${data.beneficiary.productNumber /* istanbul ignore next */ || ''}</D2>
					<R1>${data.issuerName /* istanbul ignore next */ || ''}</R1>
					<R2>${data.issuerFirstSurname /* istanbul ignore next */ || ''}</R2>
					<R3>${data.issuerSecondSurname /* istanbul ignore next */ || ''}</R3>
					<R4>${data.issuerCno /* istanbul ignore next */ || ''}</R4>
					<R5>${data.issuerPhone /* istanbul ignore next */ || ''}</R5>
					<RD1>${data.issuerAddress /* istanbul ignore next */ || ''}</RD1>
					<RD2>${data.issuerZipCode /* istanbul ignore next */ || ''}</RD2>
					<RD3>${data.issuerCity /* istanbul ignore next */ || ''}</RD3>
					<RD4>${data.issuerState /* istanbul ignore next */ || ''}</RD4>
					<RD5>${data.issuerCountry /* istanbul ignore next */ || ''}</RD5>
					<RN1>${data.issuerDoiType /* istanbul ignore next */ || ''}</RN1>
					<RN2>${data.issuerDoiNumber /* istanbul ignore next */ || ''}</RN2>
					<RN3>${data.issuerDoiExpiredDate /* istanbul ignore next */ || ''}</RN3>
					<RN4>${data.issuerDoiCountry /* istanbul ignore next */ || ''}</RN4>
					<RN5>${data.issuerBirthDate /* istanbul ignore next */ || ''}</RN5>
					<RN6>${data.issuerBirthCountry /* istanbul ignore next */ || ''}</RN6>
					<RN7>${data.issuerNationality /* istanbul ignore next */ || ''}</RN7>
				</ORDEN>
			`.replace(/(\r\n\t|\n|\r|\t)/gm, '');

			return xml;
		},

		async getQRCodesFromDeposit({ dispatch }, { depositId, mode = 'horizontal' }) {
			const deposit = await dispatch('get', { depositId });

			return dispatch('generateCodes', {
				mode,
				details: deposit.details,
			});
		},

		/* istanbul ignore next */
		generateCodes({ dispatch }, { details, mode = 'horizontal' }) {
			const rotate = {
				horizontal: 'N',
				vertical: 'R',
			}[mode];
			return Promise.all(
				details.map(async (detail, i) => {
					const canvas = document.createElement('canvas');

					const xml = await dispatch('generateCodeData', detail);

					BwipJs.render(
						{ bcid: 'pdf417', text: xml },
						BwipJs.DrawingCanvas(
							{
								bcid: 'pdf417',
								text: xml,
								paddingleft: 0,
								paddingright: 0,
								paddingtop: 0,
								paddingbottom: 0,
								textxalign: 'center',
								monochrome: true,
								backgroundcolor: 'ffffff',
								rotate,
							},
							canvas
						)
					);
					const id = `barcode-${i + 1}`;
					canvas.id = id;

					return {
						id,
						amount: detail.amount,
						base64Image: canvas.toDataURL('image/png', 1.0),
						canvas,
					};
				})
			);
		},
	},
};
