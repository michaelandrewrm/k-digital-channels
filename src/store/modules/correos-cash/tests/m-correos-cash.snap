// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`m-correos-cash return base64 qr code from a deposit: qr-codes 1`] = `
Array [
  Object {
    "amount": 2490,
    "base64Image": "data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAgAAAAC6CAYAAADYtvXAAAAABmJLR0QA/wD/AP+gvaeTAAAPbUlEQVR4nO3dy47jOg4A0NuD+f9fvrMYZGOA4ENKlws8Z9cVWVL8SBMGKf35559//v3nwL///v/wP3/+tP799Pm8228m6jdr/xwvO747TjRe9vfuuM/Pn+1uHx99Pv1+Ufvu/dSddzaP7Ljsfu7Oa9rP9H6dzuPWdb/93Gbzno4TfR6Nd+v5yuY7fe6r87z1+xaZPg+/5f+Xru73rF7P/4xmAwD8agIAAFhIAAAACwkAAGAhAQAALPTfn55ApJotmWVFVrMlu+PfyhatHj8dP8qm7WYBZ+2y/p/tp9nY1e8RZb1Oqwmm1+kpm9ezXfd6ZOcnOi5qX/3et5+j7HtF43azrLvnoXs+qtUJWT/RvE6d3gfd61J9/qtVZbfnnc3j6bS66ad5AwAACwkAAGAhAQAALCQAAICFBAAAsNDrqgC6azRPszyr2fNV1Wza59+na2pXs0+756G7lnU1Wzc6vpsdXW1fPV/V8ap/z+YTHd/deyLTrSKIjru1Jn92/m9XyVR/N7rZ4Nl4UfvsuahWPUTtu/fptJqhe3zUvlv9c2uvgml1RvV6/Jbs/w9vAABgIQEAACwkAACAhQQAALCQAAAAFnpdFUA3yzxq111rOvu8m4V/K5v79PjqfKpZ6pHumvOnulnep1ngp06zoLt7P0THP9t155GZVjdMs+ar40f9nh4X6T5f0z0pbmed36p+6o4T7dXx/Pz579NqnG51Rjbvp7dXBXgDAAALCQAAYCEBAAAsJAAAgIUEAACw0OuqAJ66azefZktn4z7HicavjtPNDu6utZ/1m7XL5ts13fugOn53D4Bp1cZ074JqtnnUT3ecbtZy5Nb5mV7P5/Hd+2O6h0H1+Ki/artp/7f2pJj+zj5Nq1C6x0Xj3tpTZFp1Eh3/Vt4AAMBCAgAAWEgAAAALCQAAYCEBAAAs9PoqgG4W/PO4j1trc59mf2dZ2lE/T9Ms5er5y8Z7jttdq3uaJdvNhs8+j87HrSzgapVGliU9XSs+c7tKoLum/mmVR6S6tnv292xep9VGp/3f3qOgOn723FT3LOhWcVS/7+leL9Pqqul8f4o3AACwkAAAABYSAADAQgIAAFhIAAAAC72uCmC6RvTz+Gr7qmlVQDWrtJqtOq1KqM6vm809rQ6I2n2rmuF0T4Zqf93qk242ejZe9fPTvTCy/qbZ76fP7+keHreer2rV0fT+rT63t/Y0mGb1n84j+97Tqpfpc/fb1/5/8gYAABYSAADAQgIAAFhIAAAACwkAAGCh11UBVNeUP12jv9quuiZ0dFw3W7Y6j6csC7iadTzNzu3OJ/p3dnw2n+neEN323eqHyOn5nt6n3ezuW3sRfHtN+Oz3o3vfZs/TtFrptFokOi5z+nuaVTOczuPZb/d8d5/rblXS6X3+Nt4AAMBCAgAAWEgAAAALCQAAYCEBAAAs9LoqgGl2bTdb/nlc1C5bO7p7XNQuU81urZ6XLHu8m/3azRo+vX7drPGsXfXzqP9qNvHtvRyqWdKnew7cUn0esizwSDcL/Hnc6XN9y+nvTre/6fPdNd0jovq7lJ2vW1Ukt/6f+WneAADAQgIAAFhIAAAACwkAAGAhAQAALPS6KoAsq/l21n6ku1Z29/Nuu+lx06z26vhZdmw1K7l7XGQ6/+516K5NXl1DfTqP0/bV+/lWdc5pNnbk9pr5z+Orz9PpXhG3svKna/5n87z1O9vdW2VaxfAc7/T/l2iev403AACwkAAAABYSAADAQgIAAFhIAAAAC72uCqCbHVztr6uaVTrNOj+dT5Ydm83zVtb7aTbzNLu6m/0+vZ9u34fTNdWnezRMTddA/8jul+qa68/PT6tFqlne3ezzTPV+vb0HyunzF33evR+72frV+X9rT4Zbe068vTrAGwAAWEgAAAALCQAAYCEBAAAsJAAAgIVeVwWQqa5BfZqlG6lmL9/KXq2uWX+ryuDZb6a7Jvi3s3arWcmn1QtR+6fp/dA9/89+pmv+35pXNG51zf+on267SPX6P/u9VXUTtcuuZzSP22vwR+Of9pM5/f3s/r9w6/cya/fWagBvAABgIQEAACwkAACAhQQAALCQAAAAFvp1VQAf316D+inLYp6uFT7Nqn6OX51nNL/TLNXqeN/ak6Cb5T7Niu+2O83+v12t8BznNMs/O+5Wtvzp5929LLpZ/5nq96xWFd3K+u/uLTE9D9Mqiunv57ev19/ak+PbvAEAgIUEAACwkAAAABYSAADAQgIAAFjodVUA2drg1azYblb+dC+BqN2trNtbexpk/XSz0aNs6uqa5dPs8+79EbWL2k+v27TqoWqa9d29r6tZ4NXzcbqG/rP/TPX6Z/M7rQbJzkc0r6ifW7Lv271futVEP509P73vo3a/Nfv/wxsAAFhIAAAACwkAAGAhAQAALCQAAICFXlcF8PGtNcJP196vjltde/zWXgDfWhs8+3s3K/90bfxbx91eKzyaT3VvimpWdjSP7v0/PR+3vnfmNLu6Os9qldGt57abRX5aRZHNv1qlk7n1+zOtRnq2r/6OTbP6b1dn/BRvAABgIQEAACwkAACAhQQAALCQAAAAFnptFUC2tnw1O7eaNZqN310j/vaa0bfXmr49r26W9DTLultFkM3zNAv4dvXF39orIJrP367OeLY7rUaYXs/q94nGyWT3XdQu+/2ruvX7cXqdb30etev+f9DN9s9+77P5vY03AACwkAAAABYSAADAQgIAAFhIAAAAC722CuCjulZ2dtztrO3qPKfVBNXxq2uMn2apZmuJZ3+vjtedT3Xt89vZv91xo2zuaLzpfdKtgsn6rZpWN5yusZ+NN90rIet3Oq9p+9PqldM17m9ls//0fVr9fTq9X6t7n/w0bwAAYCEBAAAsJAAAgIUEAACwkAAAABZ6fRXAUzfLd5rNWc2Wra7x/XRr7f3p2unV6oLuvJ7/nmb3VrP8b53vaXZ3Nk50naZVFE+3s4ynWfndNdKrWflTt9bKz65PtxolG/907fqn7vPX3aOge/+dZuFH84n66R53q3rktGrkb/EGAAAWEgAAwEICAABYSAAAAAsJAABgoddWAWRZ4dVqgO4eAlNZdv50bf7p2uTZ+ZhmyU5Ns6KfutnI1eqHaXZ193pWqxemWdbRcxK1q863O261v2+tvR+5VX1zq8rgo/qcdp/nW/Oq/r5F/d56Pm9VL1Xnl43/rWqvv8UbAABYSAAAAAsJAABgIQEAACwkAACAhV5bBTBd4z1bq7vabzUrtDqv7prw3fk822XzqR4X9TPNTs/m083Gra51HvWTzbPa/jRLPFKtcuiu+T7dQ6P6PEyz+6tZ3tU166P5dPdmyPqZVk9Unf4+TH/vplUuUbvuWvvT8TLVaqLT668KAAB4HQEAACwkAACAhQQAALCQAAAAFnpdFUA1i//ZPnOalXu6BnU3m3Y6j2i8anbzdI+Cbvb287jpmubTPQ1u7QlQzUo/rTq4lU2c7amRzePW94naV6sETp+3aF7Zc5Mdn81jWo0QuZ1tPs327/6+VvcOeP779P64XXUVzfe38AYAABYSAADAQgIAAFhIAAAACwkAAGCh11UBfGRZ/9Xs9qjfqm/tJfBsV53ft9cc/9aa4VkWb3evgKz/bG3vW+ftW9nI0+zu6vXLqi+q7aL5nn7/blb4VHY/R+NnsucjO59Z9cV0r5Sov2qV1eleI9M9M25VUWX9Vf9/uTWPn+YNAAAsJAAAgIUEAACwkAAAABYSAADAQq+rAqhmoU7X+q6qZml3x+3uaVDNpq4eV81u7Z73LJs8+nd3/tXs3ai/aNwo6z0Sna/uGunZ96+uvd/N9u7O7/S63bruUX9RP5Fb2d7dPSBO91Y43XPiVnVD93npzuO02utb3/d0/LfxBgAAFhIAAMBCAgAAWEgAAAALCQAAYKHXVQF8dLNcn+2if0/X0q5mj3bXzs6yabvZtdnn3TX2M91s726WfKa7lvztaorqPKbZ8dO9K6K/V9eoP62KOF1T/rS6Z1rFks2v+nnUvruXQnX8b1VVTPd0yO6zn7o/ur/nT9Xz/vbs/w9vAABgIQEAACwkAACAhQQAALCQAAAAFnptFUC0hnl1bfPqGvjZ+B/TNeifulneUXbwNMu0mjUffT7Nfo7G6c5nuldBNq9o/O55Ph3/9PqerjUfmWaxT/cEOM3WjuZVbR/NZ+rWmvSRbhVR9vsaHX96fqp7aJya3n/dapmo36j/t/EGAAAWEgAAwEICAABYSAAAAAsJAABgoddVAXSz+rPju58/23WzubMqhei47ryyv0+rFbpZ9tWqjGxe1XlU251m0U/XQM/uh6ifp9PqiG72/zR7/Nm+u0dC1m9X9Xt0q3qme4p8a8+Rqmjet/Y+6FZ/RONP79vn8Vm7rP2t6/D27P8PbwAAYCEBAAAsJAAAgIUEAACwkAAAABZ6XRVAd430btbodM3yLPt0uvZ0N1u0m90afT6tosiqATKn2d/dbPXp8dn8uvdRde31ruw6TKsQurrXv7sW/bQq5Keel+y6dKuabn2P53jTqoCn6Z4L3aqWrL+P6n2Stf/2Hho/zRsAAFhIAAAACwkAAGAhAQAALCQAAICFXlcFkGUDf1TX6s6ysqvziWTZ4NMs8+o4XdOs1Ol1yY7Pxnvqjl+tVrh1PrvZ1afZ11E/z/5uzWtaTRDNo6q7R0P071vZ7d01/qN+T9tNqwOy53laVVBtP62WunU+uln8UT9R+9t7OnyLNwAAsJAAAAAWEgAAwEICAABYSAAAAAu9rgpguuZz9bjumvPTNZ+zrP3p2tC3s2Cr2c3VPRWy8aLPs/Ej3az/6PPba35PqzZu33fVKohuVnZ1Plm7yLTa5bQq4zl+1n91HpHpXiDR9b1drXB7HtXxq1UBXd37qfo93r7mf8QbAABYSAAAAAsJAABgIQEAACwkAACAhV5XBdDNGq9+/rey0U+zm6vZrtWs7ltr82eyLOBovCxLv7uG+LTa42matXx7r4Os/TT7ulod8OznNDs+k9031T0BMqfVNNO1/LP+ptn4kdO177t7mlTncVp1crpnxmlVQvT3W9Vef4s3AACwkAAAABYSAADAQgIAAFhIAAAAC72uCuDj9lrgz36n/VVVqxVOs/C72bS3slSra61Xj+tm0U+zsLM14adZz9Pqlepa9dO13rtZ9N/asyKb1/O402qfbJzpc3Prd6B7nm9fh9PnrluN0b1Po+NOqy2y8U69Pev/yRsAAFhIAAAACwkAAGAhAQAALCQAAICF/gcj4pPBagHBHAAAAABJRU5ErkJggg==",
    "canvas": <canvas
      height="186"
      id="barcode-1"
      width="512"
    />,
    "id": "barcode-1",
  },
  Object {
    "amount": 20,
    "base64Image": "data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAgAAAAC6CAYAAADYtvXAAAAABmJLR0QA/wD/AP+gvaeTAAAPeUlEQVR4nO3dy47jOg4A0NuD+f9fvrMYZGOA4ENKlws8Z9cV21L8SBMGSf35559//v3nwL///n/3P3/+tP799Pm8e9xMdNxs++d42f7dcaLxsr93x31+/tzu9v7R59PvF23fvZ+6887mke2X3c/deU2PM71fp/O4dd1vP7fZvKfjRJ9H4916vrL5Tp/76jxv/b5Fps/Db/n/pav7PavX8z+j2QAAv5oAAAAWEgAAwEICAABYSAAAAAv996cnEKlmS2ZZkdVsye74t7JFq/tPx4+yabtZwNl22fGf20+zsavfI8p6nVYTTK/TUzav53bd65Gdn2i/aPvq9779HGXfKxq3m2XdPQ/d8zHNgq9WkZzen6f3Qfe6VJ//alXZ7Xln83g6rW76ad4AAMBCAgAAWEgAAAALCQAAYCEBAAAs9LoqgG6P5mmWZzV7vqqaTfv8+7SndjX7tHseur2sq9m60f7Vnt/d7avnqzpe9e/ZfKL9u2tPZLpVBNF+t3ryZ+f/dpVM9Xejmw2ejRdt363myMY/XStgWs3Q3T/avlv9c2utgurv1FP1PP+W7P8PbwAAYCEBAAAsJAAAgIUEAACwkAAAABZ6XRVAN8s82q7bazr7vJuFfyub+3T/6nyqWeqRbs/5U90s79Ms8FOnWdDdtR+i/Z/bdeeRmVY3VJ/7bH7ZfVutJjrt8d59vqZrUtzOOr9V/dQdJ1qr4/n589+n1TjV+yyb1+3n6G/xBgAAFhIAAMBCAgAAWEgAAAALCQAAYKHXVQE8dXs3n2ZLZ+M+x4nGr47TzQ7u9trPjpttl823a7r2QXX87hoA06qN6doF1Wzz6DjdcbpZy5Fb52d6PZ/7d++P6RoG1f2j41W3mx7/1poU09/Zp2kVSne/aNxba4pMq06i/d/KGwAAWEgAAAALCQAAYCEBAAAsJAAAgIVeXwXQzYJ/7vdxqzf3afZ3lqUdHedpmqVcPX/ZeM9xu726p1my3Wz47PPofNzKAq5WaWRZ0tNe8ZnbVQLdnvqnVR6Ram/37O/ZvE6rjU6Pf3uNgur42XNTXbOgW8VR/b6na71Mq6um8/0p3gAAwEICAABYSAAAAAsJAABgIQEAACz0uiqAaY/o5/7V7aumVQHVrNJqtuq0KqE6v24297Q6INruW9UMp2syVI/XrT7pZqNn41U/P10LIzveNPv99Pk9XcPj1vNVrTqa3r/V5/bWmgbTrP7TeWTfe1r1Mn3ufnvv/ydvAABgIQEAACwkAACAhQQAALCQAAAAFnpdFUC1p/xpj/7qdtWe0NF+3WzZ6jyesizgatbxNDu3O5/o39n+2Xyma0N0t+9WP0ROz/f0Pu1md99ai+DbPeGz34/ufZs9T9NqpdNqkWi/zOnvaVbNcDqP53G757v7XHerkk7v87fxBgAAFhIAAMBCAgAAWEgAAAALCQAAYKHXVQFMs2u72fLP/aLtst7R3f2i7TLV7Nbqecmyx7vZr92s4dPr180az7arfh4dv5pNfHsth2qW9OmaA7dUn4csCzzSzQJ/7nf6XN9y+rvTPd70+e6arhFR/V3KztetKpJb/8/8NG8AAGAhAQAALCQAAICFBAAAsJAAAAAWel0VQJbVfDtrP9Ltld39vLvddL9pVnt1/Cw7tpqV3N0vMp1/9zp0e5NXe6hP53G6ffV+vlWdc5qNHbndM/+5f/V5Ol0r4lZW/rTnfzbPW7+z3bVVplUMz/FO/3+J5vnbeAMAAAsJAABgIQEAACwkAACAhQQAALDQ66oAutnB1eN1VbNKp1nnp/PJsmOzed7Kej/NZp5mV3ez36f30+37cNpTfbpGw9S0B/pHdr9Ue64/Pz+tFqlmeXezzzPV+/X2Giinz1/0efd+7GbrV+f/rTUZbq058fbqAG8AAGAhAQAALCQAAICFBAAAsJAAAAAWel0VQKbag/o0SzdSzV6+lb1a7Vl/q8rgedxMtyf4t7N2q1nJp9UL0fZP0/uhe/6fx5n2/L81r2jcas//6Djd7SLV6/887q2qm2i77HpG87jdgz8a//Q4mdPfz+7/C7d+L7Pt3loN4A0AACwkAACAhQQAALCQAAAAFhIAAMBCv64K4OPbPaifsizmaa/waVb1c/zqPKP5nWapVsf71poE3Sz3aVZ8d7vT7P/b1QrPcU6z/LP9bmXLn37eXcuim/WfqX7PalXRraz/7toS0/MwraKY/n5++3r9rTU5vs0bAABYSAAAAAsJAABgIQEAACwkAACAhV5XBZD1Bq9mxXaz8qdrCUTb3cq6vbWmQXacbjZ6lE1d7Vk+zT7v3h/RdtH20+s2rXqommZ9d+/rahZ49Xyc9tB/Hj9Tvf7Z/E6rQbLzEc0rOs4t2fft3i/daqKfzp6f3vfRdr81+//DGwAAWEgAAAALCQAAYCEBAAAsJAAAgIVeVwXw8a0e4ae996vjVnuP31oL4Fu9wbO/d7PyT3vj39rvdq/waD7VtSmqWdnRPLr3//R83PremdPs6uo8q1VGt57bbhb5aRVFNv9qlU7m1u/PtBrpuX31d2ya1X+7OuOneAMAAAsJAABgIQEAACwkAACAhQQAALDQa6sAst7y1ezcatZoNn63R/ztntG3e03fnlc3S3qaZd2tIsjmeZoFfLv64m+tFRDN529XZzy3O61GmF7P6veJxslk9120Xfb7V3Xr9+P0Ot/6PNqu+/9BN9s/+73P5vc23gAAwEICAABYSAAAAAsJAABgIQEAACz02iqAj2qv7Gy/21nb1XlOqwmq41d7jJ9mqWa9xLO/V8frzqfa+/x29m933CibOxpvep90q2Cy41ZNqxtOe+xn403XSsiOO53XdPvT6pXTHve3stl/+j6t/j6d3q/VtU9+mjcAALCQAAAAFhIAAMBCAgAAWEgAAAALvb4K4Kmb5TvN5qxmy1Z7fD/d6r0/7Z1erS7ozuv572l2bzXL/9b5nmZ3Z+NE12laRfF0O8t4mpXf7ZFezcqfutUrP7s+3WqUbPzT3vVP3eevu0ZB9/47zcKP5hMdp7vfreqR06qRv8UbAABYSAAAAAsJAABgIQEAACwkAACAhV5bBZBlhVerAbprCExl2fnT3vzT3uTZ+ZhmyU5Ns6KfutnI1eqHaXZ193pWqxemWdbRcxJtV51vd9zq8b7Vez9yq/rmVpXBR/U57T7Pt+ZV/X2Ljnvr+bxVvVSdXzb+t6q9/hZvAABgIQEAACwkAACAhQQAALCQAAAAFnptFcC0x3vWq7t63GpWaHVe3Z7w3fk8t8vmU90vOs40Oz2bTzcbt9rrPDpONs/q9qdZ4pFqlUO35/t0DY3q8zDN7q9meVd71kfz6a7NkB1nWj1Rdfr7MP29m1a5RNt1e+1Px8tUq4lOr78qAADgdQQAALCQAAAAFhIAAMBCAgAAWOh1VQDVLP7n9pnTrNzTHtTdbNrpPKLxqtnN0zUKutnbz/2mPc2naxrcWhOgmpV+WnVwK5s4W1Mjm8et7xNtX60SOH3eonllz022fzaPaTVC5Ha2+TTbv/v7Wl074Pnv0/vjdtVVNN/fwhsAAFhIAAAACwkAAGAhAQAALCQAAICFXlcF8JFl/Vez26PjVn1rLYHndtX5fbvn+Ld6hmdZvN21ArLjZ729b523b2UjT7O7q9cvq76obhfN9/T7d7PCp7L7ORo/kz0f2fnMqi+ma6VEx6tWWZ2uNTJdM+NWFVV2vOr/L7fm8dO8AQCAhQQAALCQAAAAFhIAAMBCAgAAWOh1VQDVLNRpr++qapZ2d9zumgbVbOrqftXs1u55z7LJo39351/N3o2OF40bZb1HovPV7ZGeff9q7/1utnd3fqfX7dZ1j44XHSdyK9u7uwbE6doKp2tO3Kpu6D4v3XmcVnt96/uejv823gAAwEICAABYSAAAAAsJAABgIQEAACz0uiqAj26W63O76N/TXtrV7NFu7+wsm7abXZt93u2xn+lme3ez5DPdXvK3qymq85hmx0/Xroj+Xu1Rf1oVcdpT/rS6Z1rFks2v+nm0fXcther436qqmK7pkN1nP3V/dH/Pn6rn/e3Z/x/eAADAQgIAAFhIAAAACwkAAGAhAQAALPTaKoCoh3m1t3m1B342/se0B/1TN8s7yg6eZplWs+ajz6fZz9E43flM1yrI5hWN3z3Pp+OfXt/TXvORaRb7dE2A02ztaF7V7aP5TN3qSR/pVhFlv6/R/qfnp7qGxqnp/detlomOGx3/bbwBAICFBAAAsJAAAAAWEgAAwEICAABY6HVVAN2s/mz/7ufP7brZ3FmVQrRfd17Z36fVCt0s+2pVRjav6jyq251m0U97oGf3Q3Scp9PqiG72/zR7/Ll9d42E7Lhd1e/RreqZrinyrTVHqqJ531r7oFv9EY0/vW+f+2fbZdvfug5vz/7/8AYAABYSAADAQgIAAFhIAAAACwkAAGCh11UBdHukd7NGpz3Ls+zTae/pbrZoN7s1+nxaRZFVA2ROs7+72erT/bP5de+jau/1ruw6TKsQurrXv9uLfloV8lPPS3ZdulVNt77Hc7xpVcDTdM2FblVLdryP6n2Sbf/tNTR+mjcAALCQAAAAFhIAAMBCAgAAWEgAAAALva4KIMsG/qj26s6ysqvziWTZ4NMs8+o4XdOs1Ol1yfbPxnvqjl+tVrh1PrvZ1afZ19Fxnse7Na9pNUE0j6ruGg3Rv29lt3d7/EfHPd1uWh2QPc/TqoLq9tNqqVvno5vFHx0n2v72mg7f4g0AACwkAACAhQQAALCQAAAAFhIAAMBCr6sCmPZ8ru7X7Tk/7fmcZe1Pe0NXv3+1d311PtMe69X5Tnv3d9eC6Jpmq0fzqPaIv5UlHv29+xydrlkxXTsiO371fD/nEc3z6VZ1z2nv/W9VE3Wvy7Sqovt8d69f97pUVb/v23v+R7wBAICFBAAAsJAAAAAWEgAAwEICAABY6HVVAKfZyN3e193jdrOuq73np98rO27mVlb9re/bHb9abTCtEsiykbPs6Wnv/OnaCdUs5Wlv9dtVNLefk2j/aS//qFplev2qPe+j7aNxItXrXj1+NTu/e7xou+l1isbr3pfZ8bvX9228AQCAhQQAALCQAAAAFhIAAMBCAgAAWOh1VQAf3WzVj1s9+LvZ39VqgNMs3+j7dHu938pOz7KBu2sjnFZ5fGS92KvHjUyz26vzqmafd3uln/Zur25/636czqf6nJ6uxVCdRzRe9fjTaobT7Pxq1cKtapGsaqFafXBafTL19qz/J28AAGAhAQAALCQAAICFBAAAsJAAAAAW+h8eb8CXUOGbegAAAABJRU5ErkJggg==",
    "canvas": <canvas
      height="186"
      id="barcode-2"
      width="512"
    />,
    "id": "barcode-2",
  },
]
`;
